### PR TITLE
Added a paragraph about client handling unknown properties in API responses

### PR DESCRIPTION
**Pull request by agr**
Addresses https://github.com/NuGet/docs.microsoft.com-nuget/issues/1073.

### CODE REVIEW

**Comment by karann-msft** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/api/overview.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/agr-future-compat/docs/api/overview.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/api/overview?branch=pr-en-us-1075) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C10%5C1%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201810012150512197-1075%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=WK6A%2BV5%2B%2FSxC6O5QX5UL4%2F45AyLJq%2FDXpY1oMDSG%2BTg%3D&st=2018-10-01T21%3A47%3A00Z&se=2018-11-01T21%3A52%3A00Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/api/overview.md [hunk 1]
```diff
@@ -68,6 +68,10 @@ In general, all non-binary data returned by a API resource are serialized using
 returned by each resource in the service index is defined individually for that resource. For more information about
 each resource, see the topics listed above.
 
+In the future, as protocol evolves, new properties may be added to JSON responses. For the client to be future-proof,
```
Comment (joelverhagen) [inline docs/api/overview.md]: `as the protocol evolves`

File: docs/api/overview.md [hunk 2]
```diff
@@ -68,6 +68,10 @@ In general, all non-binary data returned by a API resource are serialized using
...
+the implementation should not assume that the response schema is final and cannot include extra data. All properties
+implementation does not understand should be ignored.
```
Comment (joelverhagen) [inline docs/api/overview.md]: `All properties that the implementation`

**Comment by karann-msft** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/api/overview.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/agr-future-compat/docs/api/overview.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/api/overview?branch=pr-en-us-1075) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C10%5C2%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201810022154344931-1075%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=jnKXWsiJFcvgjM%2B2tkBf43p7Oc%2BX%2BlXj27Ywc65vmCo%3D&st=2018-10-02T21%3A50%3A48Z&se=2018-11-02T21%3A55%3A48Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**